src/firefly/js/ui/dynamic/EmbeddedPositionSearchPanel.jsx

```diff
@@ -164,26 +166,33 @@ export function EmbeddedPositionSearchPanel({
         ? <WrapperComponent {...additionalProps}>{internals}</WrapperComponent>
         : internals;
 
+    const keyVal = (k, v, isLast) => (
+        <>
+            <Typography component='span' color={'primary'}>{k}: </Typography> {v}
+            {!isLast && ', '}
+        </>
+    );
+
     function SearchSummary({title, searchType, coords, radius, fileName, rows, showCoords, showRadius, showFileInfo, sx}) {
+
+        //Label/Key & Value pairs do display, calculating here to determine easily where the last comma should be
+        const keyValuePairs = [
+            { k: 'Search Type', v: searchType },
+            ...(showRadius && searchType === 'Cone' ? [{ k: 'Search Radius', v: radius }] : []),
+            ...(showCoords && searchType !== 'Multi-Object' && coords ? [{ k: 'Coordinates', v: coords }] : []),
+            ...(showFileInfo && searchType === 'Multi-Object' && fileName && rows ? [
+                { k: 'File Name', v: fileName },
+                { k: 'Rows', v: rows }
+            ] : [])
+        ];
+
         return (
             <Stack sx={sx}>
                 {title && <><Typography color={'warning'} level='title'>{title}</Typography></>}
                 <Typography color={'neutral'} level='body-md'>
-                    <Typography component='span' color={'primary'}>Search Type</Typography>: {searchType}
-                    {showRadius && searchType === 'Cone' && <>
-                        , <Typography component='span' color={'primary'}>Search Radius</Typography>: {radius}
-                    </>}
-                    {showCoords && searchType !== 'Multi-Object' && coords &&
-                        <>
-                            , <Typography component='span' color={'primary'}>Coordinates</Typography>: {coords}
-                        </>
-                    }
-                    {showFileInfo && searchType === 'Multi-Object' && fileName && rows &&
-                        <>
-                            , <Typography component='span' color={'primary'}>File Name</Typography>: {fileName}
-                            , <Typography component='span' color={'primary'}>Rows</Typography>: {rows}
-                        </>
-                    }
+                    {keyValuePairs.map((pair, index) =>
+                        keyVal(pair.k, pair.v, index === keyValuePairs.length - 1)
+                    )}
```

Comment:
I think you misunderstood.  Now, it looks worse than before.  What I meant is this,
```js
    { keyVal('Search Type', searchType) }
    { showRadius && searchType === 'Cone' && keyVal('Search Radius', radius) }
    { showCoords && searchType !== 'Multi-Object' && coords && keyVal('Coordinates', coords) }
```
If you want to insert a `,` in between, add a `<Stack divider={', '} direction='row'>` around them.  Of course, your keyVal function have to return an element and not fragment for this to work.
This is only to give you an idea of how it could have been done.  As long as it's working for you, I am ok you leaving it as is.

Reply:
Yeah this makes sense and the code is definitely cleaner using this suggested approach, but even after setting `direction='row'` for the divider stack, the formatting still looks kind of weird (I can add a screenshot if you'd like to take a look), so I'll leave this as is.